packages/ng/modal/src/lib/modal.service.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, Type } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
+import { LuModalModule } from 'dist/ng/modal';
+import { BehaviorSubject } from 'rxjs';
+import { ILuModalContent } from './modal.model';
+import { LuModal } from './modal.service';
+
+@Component({
+	selector: 'lu-test-modal',
+	standalone: true,
+	template: `Content`,
+})
+export class ModalContentComponent implements ILuModalContent {
+	title = new BehaviorSubject('OriginalTitle');
+}
+
+@Component({
+	selector: 'lu-test-opener',
+	template: `Content`,
+})
+export class ModalOpenerComponent {
+	constructor(public modal: LuModal) {}
+}
+
+describe('LuModal', () => {
+	let spectator: Spectator<ModalOpenerComponent>;
+
+	const createComponent = createComponentFactory<ModalOpenerComponent>({
+		component: ModalOpenerComponent,
+		imports: [LuModalModule],
+	});
+
+	beforeEach(() => {
+		spectator = createComponent();
+	});
+
+	function openModal<T>(type: Type<T>): T {
+		spectator.component.modal.open(type);
+		return spectator.debugElement.parent.query(By.directive(type)).componentInstance as T;
```

Comment:
@mathieu-bellange In a perfect world, I would have done:
```ts
spectator.query(type, { root: true })
```
But the `root` option only exists for string selectors.

Reply:
En attente d'une release de Spectator avec [cette PR](https://github.com/ngneat/spectator/pull/580) 😃